File: 정선/c++/programmers/나누어떨어지는숫자배열.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr, int divisor) {
    vector<int> answer;
    
    for (int i=0; i < arr.size(); i++){
        if (arr[i] % divisor == 0){
            answer.push_back(arr[i]);}
}
    if (answer.size() == 0) {
        answer.push_back(-1);
        return answer;}
    // 혹은 아래와 같이 벡터 선언할 수 있음
    // return answer.empty() ? vector<int>(1, -1) : answer;
    
    sort(answer.begin(), answer.end());
    
    return answer;
}

// ㅊㅏㅈㄱㅗ ㅈㅓㅇㄹㅕㄹㅎㅏㄴㅡㄴㄱㅔ ㅁㅏㅈㅈㅣ

File: 예림/pgs/나누어 떨어지는 숫자 배열.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int divisor) {
        Arrays.sort(arr);
        
        int[] answer = Arrays.stream(arr).filter(x -> x % divisor == 0).toArray();
        
        if(answer.length == 0) {
            answer = new int[]{-1};
        }
        
        return answer;
    }
}